membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -44,10 +46,8 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext) exten
     }
 
     val zuoraAttributesDisjunction: DisjunctionT[Future, String, Future[Option[ZuoraAttributes]]] = for {
-      accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId, identityIdToAccountIds), identityId))
-      accountIds = queryToAccountIds(accounts)
-      accountSummaries <- EitherT[Future, String, List[AccountSummary]](getResultIf(accountIds.nonEmpty, "ZuoraGetAccountSummaries", getAccountSummaries(accountIds, identityId, accountSummaryForAccountId), Nil, identityId))
-      subscriptions <- EitherT[Future, String, List[AccountWithSubscription]](getResultIf(accountSummaries.nonEmpty, "ZuoraGetSubscriptions", getSubscriptions(accountSummaries, identityId, subscriptionsForAccountId), Nil, identityId))
+      accounts <- EitherT[Future, String, GetAccountsQueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId, identityIdToAccountIds), identityId))
+      subscriptions <- EitherT[Future, String, List[AccountWithSubscription]](getResultIf(accounts.size > 0, "ZuoraGetSubscriptions", getSubscriptions(accounts, identityId, subscriptionsForAccountId), Nil, identityId))
```

Comment:
I'm still trying to figure out if getResultIf works as expected or not. It seems to me that the param FutureResult (in this case getSubscriptions(..) ) should be made lazy to make sure it doesn't execute in case the condition is false.

Reply:
agree, the future will start straight away, the condition is whether we time and log the result.  Although we don't start the timer until we enter the timing method, by which time it will have already done some of the processing.

I think this was originally added to stop cluttering up the logs as much, the name is kind of true, but I think it's quite misleading (we don't "get" the result if false, but we do calculate it regardless)